src/fu-device-list.c

```diff
@@ -138,6 +151,7 @@ fu_device_list_get_active (FuDeviceList *self)
 		FuDeviceItem *item = g_ptr_array_index (self->devices, i);
 		g_ptr_array_add (devices, g_object_ref (item->device));
 	}
+	g_ptr_array_sort (devices, fu_device_list_sort_by_priority);
```

Comment:
Not convinced about putting it in the device-list getter -- switching around the order of the internal array (when other engine things might be operating on the list already) seems a little fragile. I think the sort function would better go in `fu_engine_get_devices()` (i.e. the consumer) rather than here.

Reply:
It is protected by a mutex and the sort happens in an allocation pointer array but point taken. I'll move to engine.